stdlib/public/core/CString.swift

```diff
@@ -15,19 +15,50 @@
 import SwiftShims
 
 extension String {
+  /// Create a new `String` by copying the nul-terminated UTF-8 data
+  /// referenced by a `cString`.
+  ///
+  /// If `cString` contains ill-formed UTF-8 code unit sequences, replaces them
+  /// with replacement characters (U+FFFD).
+  ///
+  /// - Requires: `cString != nil`
+  public init(cString: UnsafePointer<CChar>) {
+    precondition(cString != nil, "cString must not be nil")
+    let len = Int(_swift_stdlib_strlen(cString))
+    let buf = UnsafeBufferPointer<UTF8.CodeUnit>(start: UnsafePointer(cString),
+      count: len)
+    self.init(codeUnits: buf, as: UTF8.self)
+  }
+
+  /// Create a new `String` by copying the nul-terminated UTF-8 data
+  /// referenced by a `cString`.
+  ///
+  /// Does not try to repair ill-formed UTF-8 code unit sequences, fails if any
+  /// such sequences are found.
+  ///
+  /// - Requires: `cString != nil`
+  public init?(validatingCString cString: UnsafePointer<CChar>) {
+    precondition(cString != nil, "cString must not be nil")
+    let len = Int(_swift_stdlib_strlen(cString))
+    let buf = UnsafeBufferPointer<UTF8.CodeUnit>(start: UnsafePointer(cString),
+      count: len)
+    self.init(validatingCodeUnits: buf, as: UTF8.self)
+  }
+}
+
+extension String {
   /// Creates a new `String` by copying the nul-terminated UTF-8 data
   /// referenced by a `CString`.
   ///
   /// Returns `nil` if the `CString` is `NULL` or if it contains ill-formed
   /// UTF-8 code unit sequences.
   @warn_unused_result
+  @available(*, deprecated, message="Use String(validatingCString:) initializer")
```

Comment:
Ah, fair point, re: recommending both.

I feel `cString: UnsafePointer<CChar>` unambiguously describes what people are used to, whereas reusing just `utf8` doesn't indicate the difference in semantics from the `codeUnits` versions (i.e., `strlen` and null avoidance). Plus it gets into initialism concerns, which is a dog I have no fight in. :stuck_out_tongue_winking_eye: